src/pipelines/alphafold_inference_pipeline.py

```diff
@@ -94,38 +109,26 @@ def alphafold_inference_pipeline(
         loop_args=run_config.outputs['model_runners'], 
         parallelism=config.PARALLELISM
         ) as model_runner:
-    model_predict = PredictOp(
+    model_predict = JobPredictOp(
+        project=project,
+        location=region,
         model_features=data_pipeline.outputs['features'],
         model_params=model_parameters.output,
         model_name=model_runner.model_name,
         prediction_index=model_runner.prediction_index,
         run_multimer_system=run_config.outputs['run_multimer_system'],
         num_ensemble=run_config.outputs['num_ensemble'],
-        random_seed=model_runner.random_seed
+        random_seed=model_runner.random_seed,
+        tf_force_unified_memory=config.TF_FORCE_UNIFIED_MEMORY,
+        xla_python_client_mem_fraction=config.XLA_PYTHON_CLIENT_MEM_FRACTION
     ).set_display_name('Predict')
-    model_predict.set_cpu_limit(config.CPU_LIMIT)
-    model_predict.set_memory_limit(config.MEMORY_LIMIT)
-    model_predict.set_gpu_limit(config.GPU_LIMIT)
-    model_predict.add_node_selector_constraint(
-        config.GKE_ACCELERATOR_KEY, config.GPU_TYPE)
-    model_predict.set_env_variable(
-        'TF_FORCE_UNIFIED_MEMORY', config.TF_FORCE_UNIFIED_MEMORY)
-    model_predict.set_env_variable(
-        'XLA_PYTHON_CLIENT_MEM_FRACTION', 
-        config.XLA_PYTHON_CLIENT_MEM_FRACTION)
 
     with dsl.Condition(is_run_relax == 'relax'):
-      relax_protein = RelaxOp(
+      relax_protein = JobRelaxOp(
+        project=project,
+        location=region,
         unrelaxed_protein=model_predict.outputs['unrelaxed_protein'],
         use_gpu=True,
+        tf_force_unified_memory=config.TF_FORCE_UNIFIED_MEMORY,
+        xla_python_client_mem_fraction=config.XLA_PYTHON_CLIENT_MEM_FRACTION
       ).set_display_name('Relax protein')
-      relax_protein.set_cpu_limit(config.RELAX_CPU_LIMIT)
-      relax_protein.set_memory_limit(config.RELAX_MEMORY_LIMIT)
-      relax_protein.set_gpu_limit(config.RELAX_GPU_LIMIT)
-      relax_protein.add_node_selector_constraint(
-        config.GKE_ACCELERATOR_KEY, config.RELAX_GPU_TYPE)
-      relax_protein.set_env_variable(
```

Comment:
I still believe using the .set_env_variable() being a good practice, unless during test those variables are not hydrated to the google-cloud-pipeline-components (v1.0.44) custom job op machine and container.

Reply:
Thanks for all the feedback @chmstimoteo. Will discuss live with you next week as discussed. Understood on the backward compatibility concern. As far as env variables: Because there's a [custom Vertex training job component](https://cloud.google.com/vertex-ai/docs/pipelines/request-gcp-machine-resources) wrapping the custom job op component (required to allow specifying machine type/GPU type), env vars specified using .set_env_variable() don't make it to the underlying component based on my testing.

Reply:
Yeah, I was also puzzled by the same problem. Last month when I was trying L4 with create_custom_training_job_from_component. The env setting by .set_env_variable() seems not working any more.  Looks like the approach in this Pull Request to set env variables is the only way.